syntax = "proto3";

package micro;

service MicroPaymentService {
  rpc StreamPayments(stream PaymentRequest) returns (stream StreamResponse);
}

message PaymentRequest {
  string session_id = 1;
  int32 station_id = 2;      // ID заправки
  float amount_cents = 3;   // Сколько литров хочет купить клиент
}

message StreamResponse {
  bool success = 1;
  int32 total_charged = 2; // в копейках
  int32 remaining_hold = 3; // в копейках
}